crud.c

```diff
@@ -0,0 +1,208 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FILE_NAME "users.txt"
+
+typedef struct {
+    int id;
+    char name[50];
+    int age;
+} Person;
+
+
+void createRecord();
+void readRecords();
+void updateRecord();
+void deleteRecord();
+int findRecordById(FILE *file, int id, Person *p);
+
+int main() {
+    int choice;
+
+    while (1) {
+        printf("\nCRUD Operations Menu:\n");
+        printf("1. Create Record\n");
+        printf("2. Read Records\n");
+        printf("3. Update Record\n");
+        printf("4. Delete Record\n");
+        printf("5. Exit\n");
+        printf("Enter your choice: ");
+        scanf("%d", &choice);
+
+        switch (choice) {
+            case 1: createRecord(); break;
+            case 2: readRecords(); break;
+            case 3: updateRecord(); break;
+            case 4: deleteRecord(); break;
+            case 5: exit(0);
+            default: printf("Invalid choice. Try again.\n");
+        }
+    }
+    return 0;
+}
+
+int is_id_unique(int id) {
+    FILE *file = fopen(FILE_NAME, "r");
+    if (!file) {
+        
+        return 1;
+    }
+
+    Person record;
+    while (fscanf(file, "%d %[^\n] %d", &record.id, record.name, &record.age) != EOF) {
+        if (record.id == id) {
+            fclose(file);
+            return 0; 
+        }
+    }
+
+    fclose(file);
+    return 1; 
+}
+
+
+void createRecord() {
+    FILE *file = fopen(FILE_NAME, "r+");

```

Comment:
The variable name can be more descriptive which provides more context about it. This applies to entire code.